### PR TITLE
[Console] Added environment variable for console color setting

### PR DESCRIPTION
**Pull request by rrehbein**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | 
| License       | MIT
| Doc PR        | TBD 

- [x] Verify tests passed

Added allowing using an environment variable to set the console color setting, superseding auto detection. Symfony 2.7 was colorless when run from cron on Amazon Linux. Symfony 3.4 was colorful in the same context.

Add something like the following to the configuration:
```xml
        <service id="console.color_listener" class="Symfony\Component\Console\EventListener\CommandColorListener">
            <tag name="kernel.event_subscriber" />
            <argument>CONSOLE_COLORS</argument>
        </service>
```

```sh
# Disable color, ignoring auto-detection
export CONSOLE_COLORS=false
# Enable color, ignoring auto-detection
export CONSOLE_COLORS=true
```

Or at the top of crontab:
```
CONSOLE_COLORS=false
```

### CODE REVIEW

**Comment by iltar** [comment]
What about adding support for `SYM_COLOR=1`?

Personally not fan of the name. What about something like `CONSOLE_COLORS`? It's odd to set something for Symfony if you'd be using Zend or Composer for example.

**Comment by rrehbein** [comment]
- Updated the suggestion of SYM_COLOR to CONSOLE_COLORS.
- Added the support for 1 as an option.

**Comment by ogizanagi** [comment]
What about just `ANSI=0|1` and updating 

https://github.com/symfony/symfony/blob/6bfc082b8751db05e32d72270989a2847b708dab/src/Symfony/Component/Console/Output/StreamOutput.php#L91-L103

and 

https://github.com/symfony/symfony/blob/6bfc082b8751db05e32d72270989a2847b708dab/src/Symfony/Component/VarDumper/Dumper/CliDumper.php#L467-L515

instead?

File: src/Symfony/Component/Console/Tests/EventListener/CommandColorListenerTest.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    /**
+     * @var string
+     */
+    protected static $varName = null;
+
+    public static function setUpBeforeClass()
+    {
+        do {
+            self::$varName = uniqid('CommandColorListener');
```
Comment (Simperfit) [inline src/Symfony/Component/Console/Tests/EventListener/CommandColorListenerTest.php]: do we really need to use uniqid here ?

Comment (rrehbein) [reply src/Symfony/Component/Console/Tests/EventListener/CommandColorListenerTest.php]: We don't need to.  Sorry, it is something from the project's standards I work on.

**Comment by rrehbein** [comment]
As is, this didn't really work out.

**Comment by ogizanagi** [comment]
@rrehbein: I think my suggestion still makes sense though. Would you provide another PR for it?

**Comment by rrehbein** [comment]
Can do.  In my development, the errors still turned colorful with the environment variable, so this listener wasn't catching the intended target as is. With changing strategy, this thread of commits basically turned to noise of do/undo.

**Pull request closed without merging**